aiida/cmdline/commands/profile.py

```diff
@@ -43,148 +49,135 @@ def complete_profiles(self, subargs_idx, subargs):
         else:
             return ""
 
-    def profile_setdefault(self, *args):
-        from aiida.common.setup import set_default_profile
 
-        if len(args) != 1:
-            print >> sys.stderr, ("Please specify the profile to be set as the default")
-            sys.exit(1)
+@verdi_profile.command("list")
+@click.option('--no-color', is_flag=True, default=False, help='to show results without colors')
+def profile_list(no_color):
+    """
+    Displays list of all available profiles.
+    """
 
-        profile = args[0]
-        # set default DB profiles
-        set_default_profile(profile, force_rewrite=True)
+    from aiida.common.setup import get_profiles_list, get_default_profile, AIIDA_CONFIG_FOLDER
+    from aiida.common.exceptions import ConfigurationError
+
+    print('Configuration folder: {}'.format(AIIDA_CONFIG_FOLDER))
+
+    try:
+        default_profile = get_default_profile()
+    except ConfigurationError as e:
+        err_msg = ("Stopping: {}\n"
+                   "Note: if no configuration file was found, it means that you have not run\n"
+                   "'verdi setup' yet to configure at least one AiiDA profile.".format(e.message))
+        echo.echo_critical(err_msg)
+
+    if default_profile is None:
+        echo.echo_error("### No default profile configured yet, run 'verdi install'! ###")
```

Comment:
if you want to return out of the function use `echo.echo_critical` and then you can remove the `return` statement